bin/copyRawDataToTmp.sh

```diff
@@ -0,0 +1,370 @@
+		#
+		project="$(basename "${sampleSheet%."${SAMPLESHEET_EXT}"}")"
+		controlFileBase="${WORKING_DIR}/logs/${project}/"
+		export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${project}.${SCRIPT_NAME}"
+
+		# shellcheck disable=SC2174
+		mkdir -m 2770 -p "${WORKING_DIR}/logs/${project}/"
+		if [[ -e "${JOB_CONTROLE_FILE_BASE}.finished" ]]
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Skipping already processed run ${project}."
+			continue
+		else
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Processing run ${project} ..."
+		fi
+		
+		printf '' > "${JOB_CONTROLE_FILE_BASE}.started"
+		
+		if [[ "${samplesheetsServerLocation}" == "localhost" ]]
+		then
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Samplesheet is on this machine"
+			if [[ -f "${WORKING_DIR}/logs/${project}/${RAWDATAPROCESSINGFINISHED}" ]]
+			then
+				log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${WORKING_DIR}/logs/${project}/${RAWDATAPROCESSINGFINISHED} present."
+				##Check if array or NGS run
+				# shellcheck disable=SC2029
+				if ssh "${samplesheetsServerLocation}" "grep 'SentrixBarcode_A' \"${sampleSheet}\""
+				then
+					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "this is array data"
+					## copy all the data from that run
+					rsync -rlptDvc "${WORKING_DIR}/./rawdata/array/GTC/${filePrefix}" "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/"
+				else		
+					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "this is NGS data"
+					## copy all the data from that run
+					rsync -rlptDvc "${WORKING_DIR}/./rawdata/ngs/${filePrefix}" "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/"
+					log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "${project} finished"
+				fi
+			else
+				log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${WORKING_DIR}/logs/${project}/${RAWDATAPROCESSINGFINISHED} absent."
+				continue
+			fi
+		else
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Samplesheet is on destination machine: ${DESTINATION_DIAGNOSTICS_CLUSTER}"
+			# shellcheck disable=SC2029
+			if ssh "${DATA_MANAGER}"@"${DESTINATION_DIAGNOSTICS_CLUSTER}" test -e "${TMP_ROOT_DIR}/logs/${project}/${project}.data.requested"
+			then
+				##Check if array or NGS run
+				if ssh "${samplesheetsServerLocation}" "grep 'SentrixBarcode_A' \"${sampleSheet}\""
+				then
+					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "this is array data"
+					## copy the requested file to the ${WORKING_DIR}
+					rsync -vt "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/logs/${project}/${project}.data.requested" "${WORKING_DIR}/logs/${project}/"
+					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/logs/${project}/${project}.data.requested present, copying will start"
+					rsyncArrayRuns "${WORKING_DIR}/logs/${project}/${project}.data.requested"
+					log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "${project} finished"
```

Comment:
Confusing statement:  this script may have finished transferring data for some project, but "project finished" may suggest some analysis has finished.